smc.s

```diff
@@ -0,0 +1,634 @@
+fill3_0:
+    
+    la $t1, smc3_0
+
+    sw $t0, 0($t1)
+
+smc3_0:
+    add $t1, $t2, $t2
+
+    la $s2, smc3_0
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_10
+    jal fail
+    
+    j fill4_10
+#----------------------------------------------------
+#
+#----------------------------------------------------
+
+#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+#new branches
+#they pointed to the next test with another gap
+#so if modification will be successful, then testing go futher without printing fail message
+new4_10:
+    j fill4_4
+new4_4:
+    j fill4_3
+new4_3:
+    j fill4_2
+new4_2:
+    j fill4_1
+new4_1:
+    j fill4_0
+new4_0:
+    j end
+
+#---------------------------------------------------
+#   test modification of branch
+#---------------------------------------------------
+
+fill4_10:
+
+    # load jump with new target at modifiable section
+    la $t1, new4_10 
+    lw $t0, 0($t1)
+    la $t1, smc4_10
+
+    sw $t0, 0($t1)
+
+    #gap
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+#modifiable code
+#--------------
+smc4_10:
+    j next1
+#--------------
+
+# if it modification haven't been successful, then print fail message
+# else go to the next section with another gap
+
+next1:
+    la $s2, smc4_10
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_4
+    jal fail
+#-----------------------------------------------
+
+fill4_4:
+    la $t1, new4_4
+    lw $t0, 0($t1)
+    la $t1, smc4_4
+
+    sw $t0, 0($t1)
+
+
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+
+smc4_4:
+    j next2
+
+next2:
+    la $s2, smc4_4
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_3
+    jal fail
+#------------------------------------------------
+fill4_3:
+    la $t1, new4_3
+    lw $t0, 0($t1)
+    la $t1, smc4_3
+
+    sw $t0, 0($t1)
+
+
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+
+smc4_3:
+    j next3
+
+next3:
+    la $s2, smc4_3
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_2
+    jal fail
+#-------------------------------------------------
+fill4_2:
+    la $t1, new4_2
+    lw $t0, 0($t1)
+    la $t1, smc4_2
+
+    sw $t0, 0($t1)
+
+
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+
+smc4_2:
+    j next4
+
+next4:
+    la $s2, smc4_2
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_1
+    jal fail
+#--------------------------------------------------
+fill4_1:
+    la $t1, new4_1
+    lw $t0, 0($t1)
+    la $t1, smc4_1
+
+    sw $t0, 0($t1)
+
+
+    add $s0, $s1, $s2
+
+
+smc4_1:
+    j next5
+
+next5:
+    la $s2, smc4_1
+    lw $s2, 0($s2)
+    beq $t0, $s2, fill4_0
+    jal fail
+#----------------------------------------------------
+fill4_0:
+    la $t1, new4_0
+    lw $t0, 0($t1)
+    la $t1, smc4_0
+
+    sw $t0, 0($t1)
+
+smc4_0:
+    j next6
+
+next6:
+    la $s2, smc4_0
```

Comment:
This code must be unreachable, right? So you may put `jal fail` here without any condition.